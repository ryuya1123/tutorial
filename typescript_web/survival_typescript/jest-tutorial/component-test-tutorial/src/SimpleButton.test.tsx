import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { SimpleButton } from "./SimpleButton";


test ("if Click button when change diplay ON or OFF", async () => {
    const user = userEvent.setup();
    render(<SimpleButton />);
    const simpleButton = screen.getByRole("button");
    expect(simpleButton).toHaveTextContent("OFF");
    await user.click(simpleButton);
    expect(simpleButton).toHaveTextContent("ON");
});

// Snapshot test
test("display 'OFF' when render soon", () => {
    const view = render(<SimpleButton />);
    expect(view.container).toMatchSnapshot();
});
