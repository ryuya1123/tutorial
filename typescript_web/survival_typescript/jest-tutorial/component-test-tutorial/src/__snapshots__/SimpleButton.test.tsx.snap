// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display 'OFF' when render soon 1`] = `
<div>
  <button>
    OFF
  </button>
</div>
`;
